lib/kino_explorer/data_transform_cell.ex

```diff
@@ -348,6 +383,15 @@ defmodule KinoExplorer.DataTransformCell do
     end
   end
 
+  defp validation_message(:filter, type, value) do
+    type = String.to_atom(type)
+
+    case cast_filter_value(type, value) do
+      {:ok, _} -> nil
+      _ -> "not #{type}"
```

Comment:
In theory it should say "not an integer" or "not a string" or "not a datetime", otherwise I believe it is gramatically incorrect. Maybe you can say "invalid value for type #{type}". This way you don't have to deal with articles. :)